src/app.js

```diff
@@ -1,196 +1,285 @@
-  button.innerHTML = "Mark";
-  button.classList.add("complete");
-  button.addEventListener("click", () => {
-    todoEl.classList.toggle("completed");
-    editButton.disabled = todoEl.classList.contains("completed");
-    saveTodoItem();
-  });
-  return button;
-};
-
-const createCommandButton = (textEl, saveTodoItem) => {
-  const button = document.createElement("button");
-  button.innerHTML = "Edit";
-  button.classList.add("edit");
-  button.addEventListener("click", () => {
-    if (button.innerHTML === "Edit") {
-      textEl.contentEditable = true;
-      textEl.focus();
-      button.innerHTML = "Save";
-      button.classList.remove("edit");
-      button.classList.add("save");
-    } else {
-      textEl.contentEditable = false;
+/**
+ * TODO:
+ * - create all elements dynamically from JS
+ * - leave only a single <div id="root"/> in html
+ * - wrap all the functionality in this module in a single function e.g. "createApp(root)"
+ * - add a new dev.js module where we can use the new `createApp()` function
+ * - call `createApp()` with the root DOM node
+ *
+ * In the test file we will be able to import the same `createApp()` function
+ * to create our application then find elements and interact with it.
+ * */
+
+export default function createApp(rootElement) {
+  const createBtn = (className, innerText) => {
+    const button = document.createElement("button");
+    button.classList.add(className);
+    button.innerHTML = innerText;
+    return button;
+  };
+
+  const createHeading = (headingSize, className, innerText) => {
+    const heading = document.createElement(headingSize);
+    heading.classList.add(className);
+    heading.innerHTML = innerText;
+    return heading;
+  };
+
+  const createHeader = () => {
+    const header = document.createElement("header");
+    const headingOne = createHeading("h1", "header", "Todo App");
+    header.appendChild(headingOne);
+    return header;
+  };
+
+  const createForm = () => {
+    const form = document.createElement("form");
+    return form;
+  };
+
+  const createInput = () => {
+    const inputElement = document.createElement("input");
+    inputElement.type = "text";
+    inputElement.classList.add("new-todo");
+    inputElement.placeholder = "Please enter todo here";
+    return inputElement;
+  };
+
+  const createFilterElement = () => {
+    const selectEl = document.createElement("select");
+    let option;
+    const inputData = "All,completed,Outstanding";
+    inputData.split(",").forEach((item) => {
+      option = document.createElement("option");
+      option.value = item.toLowerCase();
+      option.innerHTML = item;
+      selectEl.appendChild(option);
+    });
+    return selectEl;
+  };
+
+  const createTodoList = () => {
+    const divEl = document.createElement("div");
+    divEl.classList.add("container");
+    const ulEl = document.createElement("ul");
+    ulEl.classList.add("list");
+    divEl.appendChild(ulEl);
+    return divEl;
+  };
```

Comment:
these factory functions which create certain elements can be moved outside of the `createApp` function as they don't depend on variables in that scope - that'd make them reusable as well